### PR TITLE
Fix account sequence desynchronisation when error

### PR DESCRIPTION
**Pull request by NicolasMahe**
Fixes https://github.com/mesg-foundation/engine/issues/1693

This PR make the lcd and rpc clients to only increase the account sequence when the transaction is ok.

### CODE REVIEW

File: cosmos/rpc.go [hunk 1]
```diff
@@ -91,23 +91,43 @@ func (c *RPC) QueryWithData(path string, data []byte) ([]byte, int64, error) {
...
 	if err != nil {
-		c.broadcastMutex.Unlock()
 		return nil, err
 	}
+	return c.waitForTxResult(signedTx)
+}
 
-	txres, err := c.BroadcastTxSync(signedTx)
-	c.broadcastMutex.Unlock()
+func (c *RPC) buildAndBroadcastMsgNoResult(msg sdktypes.Msg) (tenderminttypes.Tx, error) {
```
Comment (NicolasMahe) [inline cosmos/rpc.go]: I split this function into 2 to be able to use a defer on the mutex unlock instead of putting it in errors.

Comment (antho1404) [reply cosmos/rpc.go:104]: you could as well explicitly unlock the mutex without a defer but I'm good with that

**Comment by antho1404** [comment]
Seems valid, could you run the perf tests and check the result on the grafana to see the impact of such a modification, the before/after visualization for let's say 1000 executions would be great

**Comment by NicolasMahe** [comment]
> Seems valid, could you run the perf tests and check the result on the grafana to see the impact of such a modification, the before/after visualization for let's say 1000 executions would be great

Same speed on the e2e tests with 10 executions:
Before:
```
        --- PASS: TestAPI/execution/many_executions_in_parallel (1.65s)
            --- PASS: TestAPI/execution/many_executions_in_parallel/create_executions (0.60s)
            --- PASS: TestAPI/execution/many_executions_in_parallel/check_in_progress (0.02s)
            --- PASS: TestAPI/execution/many_executions_in_parallel/check_completed (1.03s)
```
After:
```
        --- PASS: TestAPI/execution/many_executions_in_parallel (1.59s)
            --- PASS: TestAPI/execution/many_executions_in_parallel/create_executions (0.56s)
            --- PASS: TestAPI/execution/many_executions_in_parallel/check_in_progress (0.02s)
            --- PASS: TestAPI/execution/many_executions_in_parallel/check_completed (1.02s)
```

**Pull request merged**